### ISSUE TITLE
GetTypeMetadataName does not work on refereced projects that have multiple target framworks

### ISSUE THREAD

**Issue by JDemler** [dotnet/roslyn#20939]
**Version Used**: 
2.3.0

**Steps to Reproduce**:
Alas I do not have the time to check if this is also true for regular .net core or .net projects.

1. Create a new asp.net core project (targeting .net core) (project 1) and reference a Dummy .net core project (project 2) as workaround for Issue #20808 

2. Create a new .net standard class library (project 3). Make project 1 reference project 3

3. Let project 3 target multiple frameworks (as described [here](https://www.bartwolff.com/Blog/2017/03/20/targeting-multiple-net-platforms-in-a-single-nuget-package-w) and [here](https://docs.microsoft.com/en-us/nuget/create-packages/supporting-multiple-target-frameworks))

4. Load project 1 with MsBuildWorkspace. Get the ProjectCompilation. 
Call `GetTypeByMetadataName` with some MetadataName from project 3

**Expected Behavior**:
`GetTypeByMetadataName` returns the demanded type.

**Actual Behavior**:
`GetTypeByMetadataName` returns null. As soon as project 3 does not target multiple frameworks anymore (only .net standard) the call returns the demanded value again.
MsBuild finds the type and compiles the projects if the type is referenced. The assembly is also correctly loaded into the ProjectCompilation.

**Comment by Pilchie** [comment]
@JDemler does the resulting compilation in step 4 actually have any references?

This may be an issue with `MSBuildWorkspace` actually opening SDK projects properly.

Tagging @DustinCampbell too.

**Comment by JDemler** [comment]
Yes it has all referenced assemblies are there (including the one from project 3).
Also, calling `GetTypeByMetadataName("System.String")` works correctly.

**Comment by JDemler** [comment]
To be more clear: The problem is not, that other frameworks are targetted:
In the .csproj of project 3
Bug occurs:
`<TargetFrameworks>netstandard1.6</TargetFrameworks>`

Bug does not occur:
`<TargetFramework>netstandard1.6</TargetFramework>`
